### PR TITLE
Update binaryen and several tests for the new br_if operand order.

### PR DESCRIPTION
**Pull request by sunfishcode**
This patch updates binaryen and many of the tests to use the new br_if operand order (https://github.com/WebAssembly/spec/pull/234). It doesn't update all the auto-generated tests from the spec repo, from llvm, or from the torture tests. I assume those can be done automatically by whatever process updates them.

### CODE REVIEW

File: src/wasm-s-parser.h [hunk 1]
```diff
@@ -881,8 +877,19 @@ class SExpressionWasmBuilder {
...
     i++;
-    if (i < s.size()) {
-      ret->value = parseExpression(s[i]);
+    if (s[0]->str() == BR_IF) {
+        Expression *expr = parseExpression(s[i]);
+        i++;
+        if (i < s.size()) {
+          ret->value = expr;
+          ret->condition = parseExpression(s[i]);
+        } else if (s[0]->str() == BR_IF) {
```
Comment (binji) [inline src/wasm-s-parser.h]: always true?

**Comment by sunfishcode** [comment]
This PR looks superseded by https://github.com/WebAssembly/binaryen/pull/189 .

**Comment by kripken** [comment]
There are changes here not in the previous merged pull, like binary format changes.

**Comment by sunfishcode** [comment]
You're right. Ok, I've updated and rebased this PR now on top of your PR so it has just the additional changes.

**Comment by kripken** [comment]
lgtm if check.py passes

**Pull request merged**